test/unit/controllers/AppFooterControllerSpec.js

```diff
@@ -2,34 +2,35 @@
 /* global describe, it, inject, expect, beforeEach */
 
 describe('AppFooterController', function () {
-  var $controller, $scope, service, serviceFlag
-
   beforeEach(module('myApp.controllers'))
 
   beforeEach(function () {
-    serviceFlag = false
-    service = {
+    this.LayoutSwitchService = {
+      serviceFlag: false,
       switchLayout: function (parameter) {
-        serviceFlag = true
+        this.serviceFlag = true
       }
     }
 
+    var LayoutSwitchService = this.LayoutSwitchService
+
     inject(function (_$controller_, _$rootScope_) {
-      $controller = _$controller_
+      this.$controller = _$controller_
 
-      $scope = _$rootScope_.$new()
-      $controller('AppFooterController', {
+      this.$scope = _$rootScope_.$new()
+      var $scope = this.$scope
+      this.$controller('AppFooterController', {
         $scope: $scope,
-        LayoutSwitchService: service
+        LayoutSwitchService: LayoutSwitchService
       })
     })
   })
 
   // define tests
   it('calls the right function', function (done) {
-    expect(serviceFlag).toBe(false)
-    $scope.switchLayout(null)
-    expect(serviceFlag).toBe(true)
+    expect(this.LayoutSwitchService.serviceFlag).toBe(false)
+    this.$scope.switchLayout(null)
```

Comment:
I know, you have few influence, but `switchLayout` would indicate a kind of `toggling` to me. So I would add another expectation for a second call of it.

Argument `null` looks strange... Maybe better just omit it?